magento/pwa-studio
# [PWA-962] Apollo V3 Cache Eviction

## Description

Refactoring cache clearing of customer and cart data to use Apollo v3's `evict` and garbage cleaning.

## Related Issue







Closes [PWA-962](https://jira.corp.magento.com/browse/PWA-962)

## Acceptance

- Utilities for manual cache deletion are deprecated - deleteCacheEntry
- These calls are replaced by cache.evict(), targeting specific entries that need purged
- After checking out, all references to cart are no longer present in InMemory or persisted cache in localStorage
- After logging out, all references to cart and customer data are no longer present in InMemory or persisted cache in localStorage

### Verification Stakeholders





## Verification Steps






#### Test scenario(s) for direct fix/feature

- Sign-in as a customer
- Verify localStorage to see Apollo cache entries for customer data
- Add a new customer address (through checkout or through account)
- Add items to cart
- Verify cart data in localStorage
- Place order
- Entries for cart should be cleared (Cart will remain but with new ID)
- Sign out
- Entries for customer data should not longer be in cache

## Breaking Changes (if any)

Removed `deleteCacheEntry` file/function

## Checklist
